### PR TITLE
Rename "Schema resolving" terminology to "URI resolving"

### PR DESCRIPTION
**Pull request by awschristou**
This change renames some classes and files to more generalized terms.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lsp/core/aws-lsp-core/src/content/uriResolver.ts [hunk 1]
```diff
@@ -2,6 +2,6 @@
  * The schema request service is used to fetch schemas. If successful, returns a resolved promise with the content of the schema.
  * In case of an error, returns a rejected promise with a displayable error string.
  */
-export interface SchemaProvider {
+export interface UriResolver {
```
Comment (justinmk3) [inline lsp/core/aws-lsp-core/src/content/uriResolver.ts:5]: does the docstring need an update

File: lsp/core/aws-lsp-core/src/content/uriResolverBuilder.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
 /**
- * Creates a middleware pipeline for resolving requests for a schema.
+ * Creates a middleware pipeline for resolving requests for content from a uri.
  * Middleware handlers will be called in the order they are added, until a
  * handler returns a response. At this point, the handlers in the call stack
  * have a chance to handle/process the response.
  */
-export class SchemaProviderBuilder {
+export class UriResolverBuilder {
```
Comment (justinmk3) [inline lsp/core/aws-lsp-core/src/content/uriResolverBuilder.ts:11]: Why does a "resolver" need a "builder", can the indirection be eliminated?

**Review by justinmk3 — APPROVED** [review]
Definitely an improvement

File: lsp/core/aws-lsp-core/src/content/uriResolver.ts (hunk 1 above)
Comment (awschristou) [reply lsp/core/aws-lsp-core/src/content/uriResolver.ts:5]: Yes, thanks for that catch

File: lsp/core/aws-lsp-core/src/content/uriResolverBuilder.ts (hunk 2 above)
Comment (awschristou) [reply lsp/core/aws-lsp-core/src/content/uriResolverBuilder.ts:11]: The builder is a way of creating a middleware pipeline that chains uri resolvers together into a composite UriResolver. You can use UriResolver without the builder though. Open to suggestions for alternative names for UriResolverBuilder.

Comment (justinmk3) [reply lsp/core/aws-lsp-core/src/content/uriResolverBuilder.ts:11]: can the resolver be composable without being wrapped in a builder?

Comment (awschristou) [reply lsp/core/aws-lsp-core/src/content/uriResolverBuilder.ts:11]: Not very nicely. The code would look more complicated [in places where you would define the composition](https://github.com/aws/aws-toolkit-common/pull/510/files#diff-3bc3c866c395540756bac81fec325f2fa410073ce82467181fe68a65a723ad8fR24), and at the place where we [hand off the resolver into third party code](https://github.com/aws/aws-toolkit-common/pull/510/files#diff-5b6b83b0b1531d2a705ec63b9fd21cadcddad7c6b60375f62458a46073ccf6e2R20)

**Pull request merged**